mcs/class/corlib/System/Environment.cs

```diff
@@ -587,10 +587,6 @@ private static string ReadXdgUserDir (string config_dir, string home_dir, string
 
 			string user_dirs_path = Path.Combine (config_dir, "user-dirs.dirs");
 
-			if (!File.Exists (user_dirs_path)) {
-				return Path.Combine (home_dir, fallback);
-			}
-
```

Comment:
we should keep this check because not having the file is very common and we shouldn't be throwing an exception if we can avoid it (makes it harder to debug with first-chance exceptions turned on in the IDE or noisy if you're tracing all exceptions on the command line).